Doc/library/io.rst

```diff
@@ -143,6 +143,32 @@ High-level Module Interface
    .. versionadded:: 3.8
 
 
+.. function:: text_encoding(encoding, stacklevel=1)
+
+   This is a helper function for functions that use :func:`open` or
+   :class:`TextIOWrapper` and take ``encoding=None`` option.
+
+   This function returns *encoding* if it is not ``None`` and "locale" if
+   *encoding* is ``None``.
+
+   This function emits an :class:`EncodingWarning` if
+   ``sys.flags.warn_default_encoding`` is true. *stacklevel* specifies where
```

Comment:
Trick to get a link:

```suggestion
   :data:`sys.flags.warn_default_encoding <sys.flags>` is true. *stacklevel* specifies where
```

The warning is only emited if the encoding is None, no?